### PR TITLE
UI fixes for jira ML-260 Issue #9, #14, #18 and #21

### PR DESCRIPTION
**Pull request by fazlan-nazeem**
Do not say "login failed" when user enters only user name and but not password and try to login, instead ask to enter the password. 

Fine tune the colors of tabs in explore view - selected, not selected, disabled. Preferably hide disabled tbs    

Need to figure out whether description is mandatory in projects and datasets (now it's inconsistent).    

No need to have close button on failed models description (in the expanded view of accordion).

### CODE REVIEW

File: apps/ml/site/analysis/analysis.jag [hunk 1]
```diff
@@ -688,6 +692,14 @@ include("../../includes/tenantAware.jag");
         return algorithmName;
     }
 
+    function removeCloseButtonFromDescription(failedModels){
+
+        for(i=0; i<failedModels.length ; i++){
```
Comment (cdathuraliya) [inline apps/ml/site/analysis/analysis.jag]: Please format code

File: apps/ml/site/data/explore.jag [hunk 2]
```diff
@@ -619,27 +619,19 @@ function selectActiveTab() {
...
-	    $('#trellis-chart-li').attr('class', 'disabled');
-        $('#trellis-chart-li').attr('title', 'Minimum of one categorical feature required to draw trellis charts');
-        $('#trellis-chart-link').attr('data-toggle', '');
+	    $('#trellis-chart-li').remove();
 	}
 	if(numericalFeatureNames.length < 2) {
-		$('#cluster-diagram-li').attr('class', 'disabled');
-        $('#cluster-diagram-li').attr('title', 'Minimum of two numerical features required for cluster diagrams');
-        $('#cluster-diagram-link').attr('data-toggle', '');
+		$('#cluster-diagram-li').remove();
```
Comment (cdathuraliya) [inline apps/ml/site/data/explore.jag]: Please format code

File: apps/ml/site/explore/explore.jag [hunk 3]
```diff
@@ -703,27 +703,19 @@ function selectActiveTab() {
 	}
 }
 
-// disable tabs based on the features and show tooltips
-function disableTabs() {
+// remove unsupported tabs 
+function removeTabs() {
     if (!(numericalFeatureNames.length >1 & categoricalFeatureNames.length >0)) {
```
Comment (cdathuraliya) [inline apps/ml/site/explore/explore.jag]: Logical AND operator is &&

File: apps/ml/site/explore/explore.jag [hunk 4]
```diff
@@ -703,27 +703,19 @@ function selectActiveTab() {
...
-        $('#scatter-plot-link').attr('data-toggle', '');
+        $('#scatter-plot-li').remove();
 	}
 	if(categoricalFeatureNames.length < 2) {
-	    $('#parallel-sets-li').attr('class', 'disabled');
-        $('#parallel-sets-li').attr('title', 'Minimum of two categorical features required for parallel sets');
-        $('#parallel-sets-link').attr('data-toggle', '');
+	    $('#parallel-sets-li').remove();
 	}
 	if(!(categoricalFeatureNames.length > 0 & numericalFeatureNames.length > 0)) {
```
Comment (cdathuraliya) [inline apps/ml/site/explore/explore.jag]: Logical AND operator is &&

File: apps/ml/site/analysis/analysis.jag [hunk 5]
```diff
@@ -688,6 +692,14 @@ include("../../includes/tenantAware.jag");
         return algorithmName;
     }
 
+    function removeCloseButtonFromDescription(failedModels){
```
Comment (Kishanthan) [inline apps/ml/site/analysis/analysis.jag:695]: Formatting issues.

File: apps/ml/site/analysis/analysis.jag [hunk 6]
```diff
@@ -688,6 +692,14 @@ include("../../includes/tenantAware.jag");
...
+
+        for(var i = 0; i < failedModels.length; i++){
```
Comment (Kishanthan) [inline apps/ml/site/analysis/analysis.jag:697]: Formatting issues.

File: apps/ml/site/data/explore.jag [hunk 7]
```diff
@@ -619,27 +619,19 @@ function selectActiveTab() {
...
-	    $('#parallel-sets-li').attr('class', 'disabled');
-        $('#parallel-sets-li').attr('title', 'Minimum of two categorical features required for parallel sets');
-        $('#parallel-sets-link').attr('data-toggle', '');
+	    $('#parallel-sets-li').remove();
 	}
-	if(!(categoricalFeatureNames.length > 0 & numericalFeatureNames.length > 0)) {
-	    $('#trellis-chart-li').attr('class', 'disabled');
-        $('#trellis-chart-li').attr('title', 'Minimum of one categorical feature required to draw trellis charts');
-        $('#trellis-chart-link').attr('data-toggle', '');
+	if(!(categoricalFeatureNames.length > 0 && numericalFeatureNames.length > 0)) {
```
Comment (Kishanthan) [inline apps/ml/site/data/explore.jag:630]: Formatting issues.

File: apps/ml/site/explore/explore.jag [hunk 8]
```diff
@@ -703,27 +703,19 @@ function selectActiveTab() {
...
-	    $('#parallel-sets-li').attr('class', 'disabled');
-        $('#parallel-sets-li').attr('title', 'Minimum of two categorical features required for parallel sets');
-        $('#parallel-sets-link').attr('data-toggle', '');
+	    $('#parallel-sets-li').remove();
 	}
-	if(!(categoricalFeatureNames.length > 0 & numericalFeatureNames.length > 0)) {
-	    $('#trellis-chart-li').attr('class', 'disabled');
-        $('#trellis-chart-li').attr('title', 'Minimum of one categorical feature required to draw trellis charts');
-        $('#trellis-chart-link').attr('data-toggle', '');
+	if(!(categoricalFeatureNames.length > 0 && numericalFeatureNames.length > 0)) {
```
Comment (Kishanthan) [inline apps/ml/site/explore/explore.jag:714]: Formatting issues.

File: apps/ml/site/home/login.jag [hunk 9]
```diff
@@ -141,7 +141,12 @@ $('#password').keydown(function(e){
 function authenticate(){
 var user = $('#userName').val();
 var passwd = $('#password').val();
-$('.wr-validation-summary').html('<p>Login failed, please check your user name and password.</p>');
+
+if(user.length >0 && passwd.length == 0){ // If user enters username but not password
```
Comment (Kishanthan) [inline apps/ml/site/home/login.jag:145]: Formatting issues.

File: apps/ml/site/project/create_project.jag [hunk 10]
```diff
@@ -235,10 +234,9 @@ $( document ).ready(function() {
         var datasetName = $( "#datasets option:selected" ).text();
         
         var projectNameValidation = isValidProjectName(projectName);
-        var descriptionValidation = isValidDescription(projectDes);
         var datasetValidation = isValidDataset(datasetName);
 
-        if(projectNameValidation && descriptionValidation && datasetValidation){
+        if(projectNameValidation  && datasetValidation){
```
Comment (Kishanthan) [inline apps/ml/site/project/create_project.jag:239]: Formatting issues.

Comment (fazlan-nazeem) [inline apps/ml/site/project/create_project.jag:239]: fixed with PR https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/home/login.jag (hunk 9 above)
Comment (fazlan-nazeem) [inline apps/ml/site/home/login.jag:145]: fixed with PR https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/explore/explore.jag (hunk 8 above)
Comment (fazlan-nazeem) [inline apps/ml/site/explore/explore.jag:714]: fixed with PR https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/data/explore.jag (hunk 7 above)
Comment (fazlan-nazeem) [inline apps/ml/site/data/explore.jag:630]: fixed with PR https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/analysis/analysis.jag (hunk 6 above)
Comment (fazlan-nazeem) [inline apps/ml/site/analysis/analysis.jag:697]: fixed with PR https://github.com/wso2/carbon-ml/pull/202

File: apps/ml/site/analysis/analysis.jag (hunk 5 above)
Comment (fazlan-nazeem) [inline apps/ml/site/analysis/analysis.jag:695]: fixed with PR https://github.com/wso2/carbon-ml/pull/202

**Pull request merged**